drivers/sensor/icm42605/icm42605.c

```diff
@@ -0,0 +1,463 @@
+				LOG_ERR("Incorrect fullscale value");
+				return -EINVAL;
+			} else {
+				drv_data->accel_sf = val->val1;
+			}
+		} else {
+			LOG_ERR("Not supported ATTR");
+			return -ENOTSUP;
+		}
+
+		break;
+	case SENSOR_CHAN_GYRO_X:
+	case SENSOR_CHAN_GYRO_Y:
+	case SENSOR_CHAN_GYRO_Z:
+	case SENSOR_CHAN_GYRO_XYZ:
+		if (attr == SENSOR_ATTR_SAMPLING_FREQUENCY) {
+			if (val->val1 > 8000 || val->val1 < 12) {
+				LOG_ERR("Incorrect sampling value");
+				return -EINVAL;
+			} else {
+				drv_data->gyro_hz = val->val1;
+			}
+		} else if (attr == SENSOR_ATTR_FULL_SCALE) {
+			if (val->val1 < GYRO_FS_2000DPS ||
+			    val->val1 > GYRO_FS_15DPS) {
+				LOG_ERR("Incorrect fullscale value");
+				return -EINVAL;
+			} else {
+				drv_data->gyro_sf = val->val1;
+			}
+		} else {
+			LOG_ERR("Not supported ATTR");
+			return -EINVAL;
+		}
+		break;
+	default:
+		LOG_ERR("Not support");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int icm42605_attr_get(const struct device *dev,
+			     enum sensor_channel chan,
+			     enum sensor_attribute attr,
+			     struct sensor_value *val)
+{
+	const struct icm42605_data *drv_data = dev->data;
+
+	__ASSERT_NO_MSG(val != NULL);
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+	case SENSOR_CHAN_ACCEL_Y:
+	case SENSOR_CHAN_ACCEL_Z:
+	case SENSOR_CHAN_ACCEL_XYZ:
+		if (attr == SENSOR_ATTR_SAMPLING_FREQUENCY) {
+			val->val1 = drv_data->accel_hz;
+		} else if (attr == SENSOR_ATTR_FULL_SCALE) {
+			val->val1 = drv_data->accel_sf;
+		} else {
+			LOG_ERR("Not supported ATTR");
+			return -EINVAL;
+		}
+
+		break;
+	case SENSOR_CHAN_GYRO_X:
+	case SENSOR_CHAN_GYRO_Y:
+	case SENSOR_CHAN_GYRO_Z:
+	case SENSOR_CHAN_GYRO_XYZ:
+		if (attr == SENSOR_ATTR_SAMPLING_FREQUENCY) {
+			val->val1 = drv_data->gyro_hz;
+		} else if (attr == SENSOR_ATTR_FULL_SCALE) {
+			val->val1 = drv_data->gyro_sf;
+		} else {
+			LOG_ERR("Not supported ATTR");
+			return -EINVAL;
+		}
+
+		break;
+
+	default:
+		LOG_ERR("Not support");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int icm42605_data_init(struct icm42605_data *data,
+			      const struct icm42605_config *cfg)
+{
+	data->accel_x = 0;
+	data->accel_y = 0;
+	data->accel_z = 0;
+	data->temp = 0;
+	data->gyro_x = 0;
+	data->gyro_y = 0;
+	data->gyro_z = 0;
+	data->accel_hz = cfg->accel_hz;
+	data->gyro_hz = cfg->gyro_hz;
+
+	data->accel_sf = cfg->accel_fs;
+	data->gyro_sf = cfg->gyro_fs;
+
+	data->tap_en = false;
+	data->sensor_started = false;
+
+	return 0;
+}
+
+
+static int icm42605_init(const struct device *dev)
+{
+	struct icm42605_data *drv_data = dev->data;
+	const struct icm42605_config *cfg = dev->config;
+
+	drv_data->spi = device_get_binding(cfg->spi_label);
+	if (!drv_data->spi) {
+		LOG_ERR("SPI device not exist");
+		return -ENODEV;
+	}
+
+	drv_data->spi_cs.gpio_dev = device_get_binding(cfg->gpio_label);
+	if (!drv_data->spi_cs.gpio_dev) {
+		LOG_ERR("GPIO device not exist");
+		return -ENODEV;
+	}
+
+	drv_data->spi_cs.gpio_dev = device_get_binding(cfg->gpio_label);
```

Comment:
duplecated gpio_level load is incorrect. I will fix.
And for CS, I followed other snesor's source code.
Without CS setting, sensor data doesn't appear when turn on sensor.
Anyway, I will try again without CS configuration.

Reply:
Removed duplicated gpi_level code.
BTW, without CS, cannot get data from sensor
I testing on Nordic nRF52 board and if you know good example code, please inform me. I will check.

Reply:
@MaureenHelm Any opinion about this change request??

Reply:
> You already did this in line 391, why are you doing it again?
> 
> Also, the chip select may not always be a GPIO, it could be managed by the SPI controller.

Need your opinion.